### PR TITLE
Adds VideoDecoder

### PR DESCRIPTION
**Pull request by rachelandrew**
Another WebCodec API: `VideoDecoder`.

Reviewer: @jpmedley

### CODE REVIEW

File: files/en-us/web/api/videodecoder/index.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+tags:
+  - API
+  - Interface
+  - Reference
+  - VideoDecoder
+browser-compat: api.VideoDecoder
+---
+{{securecontext_header}}{{DefaultAPISidebar("WebCodecs API")}}
+
+The **`VideoDecoder`** interface of the {{domxref('WebCodecs API','','',' ')}} decodes chunks of video.
```
Comment (jpmedley) [inline files/en-us/web/api/videodecoder/index.md:13]:
```suggestion
The **`VideoDecoder`** interface of the {{domxref('WebCodecs API','','','true')}} decodes chunks of video.
```

Comment (jpmedley) [reply files/en-us/web/api/videodecoder/index.md:13]: This makes the title not monospaced.

File: files/en-us/web/api/videodecoder/index.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+- {{domxref("VideoDecoder.VideoDecoder()")}}
+  - : Creates a new `VideoDecoder` object.
+
+## Properties
+
+- {{domxref("VideoDecoder.decodeQueueSize")}}{{ReadOnlyInline}}
+  - : An integer representing the number of decode queue requests.
+- {{domxref("VideoDecoder.state")}}{{ReadOnlyInline}}
+  - : Represents the state of the underlying codec and whether it is configured for decoding.
```
Comment (jpmedley) [inline files/en-us/web/api/videodecoder/index.md:25]:
```suggestion
  - : Indicates whether the underlying codec is configured for decoding.
```

File: files/en-us/web/api/videodecoder/videodecoder/index.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+tags:
+  - API
+  - Constructor
+  - Reference
+  - VideoDecoder
+browser-compat: api.VideoDecoder.VideoDecoder
+---
+{{securecontext_header}}{{DefaultAPISidebar("WebCodecs API")}}
+
+The **`VideoDecoder()`** constructor creates a new {{domxref("VideoDecoder")}} object with the provided `init.output` callback assigned as the output callback, the provided `init.error` callback as the error callback, and the {{domxref("VideoDecoder.state")}} set to `"unconfigured"`.
```
Comment (jpmedley) [inline files/en-us/web/api/videodecoder/videodecoder/index.md:13]: I'm not crazy about describing the init members in the summary when they are described in a dedicated section. What's the argument for it?

File: files/en-us/web/api/videodecoder/videodecoder/index.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+## Syntax
+
+```js
+new VideoDecoder();
+```
+
+### Parameters
+
+- `init`
+  - : A dictionary object containing two required callbacks.
```
Comment (jpmedley) [inline files/en-us/web/api/videodecoder/videodecoder/index.md:24]:
```suggestion
  - : A dictionary object containing two callbacks.
```

File: files/en-us/web/api/videodecoder/videodecoder/index.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    - `output`
+      - : A callback which is passed a single argument of an {{domxref("VideoFrame")}} object.
```
Comment (jpmedley) [inline files/en-us/web/api/videodecoder/videodecoder/index.md:26]:
```suggestion
      - : A callback which takes a {{domxref("VideoFrame")}} object as its only argument.
```

File: files/en-us/web/api/videodecoder/videodecoder/index.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    - `error`
+      - : A callback which is passed a single argument of the error that has been thrown.
```
Comment (jpmedley) [inline files/en-us/web/api/videodecoder/videodecoder/index.md:28]:
```suggestion
      - : A callback which takes an {{jsxref("Error")}} object as its only argument.
```

File: files/en-us/web/api/videodecoder/videodecoder/index.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+## Examples
+
+In the following example an `VideoDecoder` is created with the two required callback functions, one to deal with the decoded frame and the other to handle errors.
```
Comment (jpmedley) [inline files/en-us/web/api/videodecoder/videodecoder/index.md:32]:
```suggestion
In the following example a `VideoDecoder` is created with the two required callback functions, one to deal with the decoded frame and the other to handle errors.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/videodecoder/configure/index.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+```
+
+### Parameters
+
+- `config`
+  - : A dictionary object containing the following members:
+    - `codec`
+      - : A {{domxref("DOMString","string")}} containing a [valid codec string](https://www.w3.org/TR/webcodecs-codec-registry/#audio-codec-registry).
+    - `description`{{Optional_Inline}}
+      - : A {{domxref("BufferSource")}} containing a sequence of codec specific bytes, commonly known as extradata.
```
[the remark this answers is not in the document]
Comment (rachelandrew) [reply files/en-us/web/api/videodecoder/configure/index.md:29]: yep.

File: files/en-us/web/api/videodecoder/videodecoder/index.md (hunk 3 above)
Comment (rachelandrew) [reply files/en-us/web/api/videodecoder/videodecoder/index.md:13]: I feel these intros should be a summary of what's going on I guess, but I'm not altogether bothered either way. I don't think we describe anywhere else that the video `state` is set to unconfigured which is important as you have to both create the decoder and configure it, where with some things you create an object and it's ready to go.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**